store/index.js

```diff
@@ -0,0 +1,13 @@
+const defaultTimeZone = 'PDT'
+
+export const state = () => ({
+  userTimeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
```

Comment:
⚠️ Hey there! If I'm not wrong, the way this is implemented will always set the `userTimezone` to the local time where the server generates the page (GitHub in this case). When server-side rendering happens, it will serialize the final state of the store and embed that to the page.

If we want to auto-detect the user's local timezone, we should try to get this on a component's mount callback (maybe on the layout?) or [creating a tiny plugin](https://nuxtjs.org/examples/plugins-client-only/) - probably moving this logic into a Vuex action. 

Of course this could mean a tiny flicker until the the client-side code picks up (showing the server's timezone for a moment) but I think it's a good tradeoff.

Another thing we could consider is storing the timezone on localStorage after a user switches it and then reusing it when the page gets refreshed. The same thing would apply: you need to check it client side.

I'm up for a quick pair session if you want to talk about it!

Reply:
You're completely right, I was so focussed on solving the date formats and the detecting the time zone that I missed that they way it's currently implemented is not being evaluated in the user's browser. I've just seen this issue in Netlify as it's printing UTC instead of Europe/Madrid. 🤦‍♀️

I created the draft PR and I wanted to share it because it's getting big, so you could take a quick look, but it's not finished yet.

Thanks a lot for the feedback! I know about the client only plugins, but I'll ping you if I need some help. ☺️

Reply:
Fantastic Vero! I just wanted to take a look and help you before this became an issue ;)